cgsmiles/pysmiles_utils.py

```diff
@@ -83,6 +83,16 @@ def rebuild_h_atoms(mol_graph, keep_bonding=False):
             ref_node = next(mol_graph.neighbors(node))
             mol_graph.nodes[node]["fragid"] = mol_graph.nodes[ref_node]["fragid"]
             mol_graph.nodes[node]["fragname"] = mol_graph.nodes[ref_node]["fragname"]
+        if mol_graph.nodes[node].get("element", "*") == "H":
+            anchor = list(mol_graph.neighbors(node))[0]
+            # the weight for the hydrogen atom was explicitly set
+            hweights = mol_graph.nodes[anchor].get('hweight', [])
+            if hweights:
+                weight = hweights.pop()
+            # make sure the weights are copied for implicit h-atoms
+            else:
+                weight = mol_graph.nodes[anchor].get("w", 1)
+            mol_graph.nodes[node]["w"] = weight
```

Comment:
@pckroon I think this bit should probably propagate all attributes to hydrogen atoms?!